cmd/client/resources/object.go

```diff
@@ -90,6 +91,68 @@ func GetAllObject(cmd *cobra.Command) error {
 	return nil
 }
 
+// EditObject edit an object.
+func EditObject(cmd *cobra.Command, args *general.ArgInfo, kind string) error {
+	getErr := func(err error) error {
+		return general.ErrorMsg(general.EditCmd, err, kind, args.Name)
+	}
+
+	// get old yaml and save it to a temp file
+	var oldYaml string
+	var err error
+	if oldYaml, err = getObjectYaml(args.Name); err != nil {
+		return getErr(err)
+	}
+	filePath := getResourceTempFilePath(kind, args.Name)
+	newYaml, err := editResource(oldYaml, filePath)
+	if err != nil {
+		return getErr(editErrWithPath(err, filePath))
+	}
+	if newYaml == "" {
+		return nil
+	}
+
+	_, newSpec, err := general.CompareYamlNameKind(oldYaml, newYaml)
+	if err != nil {
+		return getErr(editErrWithPath(err, filePath))
+	}
+	err = ApplyObject(cmd, newSpec)
+	if err != nil {
+		return getErr(editErrWithPath(err, filePath))
+	}
+	os.Remove(filePath)
```

Comment:
move this line into `editResource` and use `defer` to ensure the file is deleted. there are other cases.